### ISSUE TITLE
Important: User information disclosed

### ISSUE THREAD

**Issue by digitalhuman** [diaspora/diaspora#6578]
The switch: 'Do not allow users to find me' does not work anymore. I am able to find me on diasp.eu even when my profile should not allow it.

Diasp.eu runs: 0.5.4.0-pbf1efb28
We (pod.readme.is) run: 0.5.4.0-pbdc55f05

Discovered by a user on our pod.

**Comment by jhass** [comment]
Did you search by diaspora id perhaps? Or are you sharing with the account you searched with?

**Comment by digitalhuman** [comment]
No just username (digitalhuman). And no, i don't share with my other account.

**Comment by jhass** [comment]
@diasp could you please check if the switch (`searchable` in the `profiles` table) is indeed set to `false` for this account on diasp.eu?

**Comment by digitalhuman** [comment]
It is false, thats why i created it to validate what one of our users said/told me.

**Comment by jhass** [comment]
So you searched for "digitalhuman" on `pod.readme.is` and `digitalhuman@diasp.eu` showed up, while in `pod.readme.is`'es `profiles` table `searchable` is `false` for that account?

**Comment by digitalhuman** [comment]
I searched on: pod.readme.is for the user 'digitalhuman' and digitalhuman@diasp.eu showed up. But, digitalhuman@diasp.eu has 'searchable' set to false.

**Comment by jhass** [comment]
This is an important detail, did you look at the `profiles` table of `pod.readme.is`?

**Comment by digitalhuman** [comment]
Well i wasn't search for myself......... as in. Me: digitalhuman@pod.readme.is, searches for user: digitalhuman@diasp.eu.
DigitalHuman@diasp.eu shouldn't be 'searchable' because this is set to false on his profile.

**Comment by jhass** [comment]
That doesn't answer my question, I'm asking whether the entry in the `profiles` table of the pod `pod.readme.is` where `diaspora_handle` is `digitalhuman@diasp.eu`, has `searchable` set to `false`.

**Comment by digitalhuman** [comment]
Explain how the handle digitalhuman@diasp.eu can exists on: pod.readme.is?

**Comment by jhass** [comment]
Please run the following queries against your database and show the result:

```sql
SELECT diaspora_handle, searchable FROM profiles WHERE diaspora_handle = 'digitalhuman@diasp.eu';
SELECT people.diaspora_handle, searchable FROM profiles JOIN people ON person_id = people.id WHERE people.diaspora_handle = 'digitalhuman@diasp.eu';
```

**Comment by digitalhuman** [comment]
MariaDB [diaspora_production]> SELECT diaspora_handle, searchable FROM profiles WHERE diaspora_handle = 'digitalhuman@diasp.eu';
Empty set (0.05 sec)

MariaDB [diaspora_production]> SELECT people.diaspora_handle, searchable FROM profiles JOIN people ON person_id = people.id WHERE people.diaspora_handle = 'digitalhuman@diasp.eu';
+-----------------------+------------+
| diaspora_handle       | searchable |
+-----------------------+------------+
| digitalhuman@diasp.eu |          0 |
+-----------------------+------------+
1 row in set (0.00 sec)

MariaDB [diaspora_production]>

**Comment by jhass** [comment]
Mmh, not what I hoped for. I'm running PostgreSQL and can't reproduce the issue, I wonder if it's a MariaDB/MySQL only bug?

**Comment by digitalhuman** [comment]
True, donno really. I know 1 thing. It was working before. Thats 100% sure.

**Comment by digitalhuman** [comment]
The thing is; it differs around other pods as well. So.... could be code, could be database. Same things happen on: diaspora.subsignal.org, joindiaspora.com.  So....

**Comment by jhass** [comment]
Well, the initial scenario I had in mind is that a pod fetched the profile while it had searchable set to true, but never established a relationship that would make it eligible for profile updates, like one contact sharing with or being followed. In that case a change from searchable true to searchable false wouldn't be propagated to said pod. That's a design issue I see no real workaround for.

**Comment by digitalhuman** [comment]
Ok, can you explain how the search works? Does it send a request to diasp.eu to handover all users like 'digitalhuman' ? or is it stored on pod.readme.is? if so, then it sucks because settings can change and if stored on pod.readme.is they need to be updated.
Better was to ask diasp.eu to handover all users with the name 'digitalhuman' and check the profile settings before handing over information.

**Comment by jhass** [comment]
It's local unless you search for specific diaspora id. There's no central authority so we have nobody else to ask to when you search for "digitalhuman".

**Comment by digitalhuman** [comment]
well you can combine the 2. Keep the search as is, but validation through request on specific pod if searchable is still true.

**Comment by digitalhuman** [comment]
Also, is it every updated ? as in, are the local stored users updated once in a while? or never?

**Comment by jhass** [comment]
So when I search for "John" I can easily contact 100 pods on each search, some of them slow to respond or even down.. The search would take 10 minutes and more each time.

Profiles are updated with started sharing requests as well as when you edit the profile it's send to all contacts.

**Comment by digitalhuman** [comment]
Contacts as in 'followers' ?

**Comment by jhass** [comment]
Both, people you follow and that follow you.

**Comment by digitalhuman** [comment]
Yeah. That is of course an issue. The search is a very difficult thing in this case. If i would like to stay anonymous and the profile is default on 'searchable' then my anonimity is at stake.

**Comment by theworldbright** [comment]
@digitalhuman I'm not sure if you still have the same settings up but I searched for `digitalhuman` from https://kentshikama.com from an account that is not sharing with both of your `digitalhuman` accounts and your `digitalhuman@diasp.eu` account did NOT show up. Your `digitalhuman@pod.readme.is` account did showed up though (I suppose you have that set to searchable?).

Of course, directly searching for `digitalhuman@diasp.eu` shows your `digitalhuman@diasp.eu` account.

I've reproduced your query on kentshikama.com's MySQL database (although I guess this isn't the issue at hand):

```
SELECT people.diaspora_handle, searchable FROM profiles JOIN people ON person_id = people.id WHERE people.diaspora_handle = 'digitalhuman@diasp.eu';
```

also gives

```
digitalhuman@diasp.eu | 0 |
```
